File: futaba_common_sample.js
/*Futaba librarryを利用したスクリプト*/

const futaba = require('./futaba.js');
const fs = require('fs');

let client = new futaba();
let fu = new futabaUtility();
fu.setTargetBuilding(["R90/research", "R90/east"])
  .setDownloadFolderPath(__dirname + '/download/');

const dir_conf = {
  depth: null
}

/**
 * 共有データAPI サンプル
 */

// 1.共有データ追加
async function Sample_1() {
  const values = [
    "{\"userProp1\": \"abc\", \"userProp2\": 1}",
    "{\"userProp1\": \"def\", \"userProp2\": 2}"
  ]
  client.setSharedData(2, fu.getTargetBuilding()[0], values)
    .then(res => console.dir(res, dir_conf));
};

// 2.共有データ検索
async function Sample_2(filter) {
  client.getSharedData(2, fu.getTargetBuilding()[0], filter)
    .then(res => console.dir(res, dir_conf));
};

// 3.共有データ削除
async function Sample_3(filter) {
  client.deleteSharedData(2, fu.getTargetBuilding()[0], filter)
    .then(res => console.dir(res, dir_conf));
};

async function main() {
  // トークンの発行・更新
  let obj = JSON.parse(fs.readFileSync('./data/config.json', 'utf8'));
  await client.getAccessToken(obj)
    .then(res => {
      obj.access_token = res.accessToken;
      obj.refresh_token = res.refreshToken;
      if (obj.access_token && obj.refresh_token) {
        fs.writeFileSync('./data/config.json', JSON.stringify(obj));
      }
      console.log(res);
    });

  //既存トークンのセット
  // client.setAccessToken(obj);

  const filter = {
    startOffset: 10,
    count: 1,
    startTime: "2020-02-01T00:00+09:00",
    endTime: "2020-03-01T00:00+09:00",
    query: [{
      path: "$.userProp1",
      op: "=",
      value: "abc"
    }]
  }

  await Sample_1();
}


main();

File: futaba_common_sample.py
import os
import futaba
import futaba_utility
import json

json_open = open('./data/config.json', 'r')
obj = json.load(json_open)
client = futaba.Futaba()
fu = futaba_utility.FutabaUtility()
fu.setTargetBuilding(["Shinsuna/TKCentralSouth"]
).setDownloadFolderPath(os.path.dirname(__file__) + "download")

def updateToken():
    token = client.getAccessToken(obj)
    with open('./data/config.json', 'w') as f:
        obj['access_token'] = token['accessToken']
        obj['refresh_token'] = token['refreshToken']
        json.dump(obj, f, indent=2)
    print(token)


# デジタルツインAPI サンプル


# 1.共有データ追加
def Sample_1():
    values = ["{\"userProp1\": \"abc\", \"userProp2\": 1}",
              "{\"userProp1\": \"def\", \"userProp2\": 2}"]
    add_value = json.dumps(values)
    response_1 = client.setSharedData(4, fu.getTargetBuilding()[0], add_value)
    print(json.dumps(response_1, indent=2))


# 2.共有データ検索
def Sample_2(filter):
    response_2 = client.getSharedData(4, fu.getTargetBuilding()[0])
    print(json.dumps(response_2, indent=2))


# 3.共有データ削除
def Sample_3(filter):
    response_3 = client.deleteSharedData(4, r90.getTargetBuilding()[0], filter)
    print(json.dumps(response_3, indent=2))


def main():
    # トークンの発行・更新++++
    # updateToken()

    # 既存トークンのセット
    client.setAccessToken(obj)

    filter = {
        "startOffset": 10,
        "count": 1,
        "startTime": "2020-02-01T00:00+09:00",
        "endTime": "2020-03-01T00:00+09:00",
        "query": [
            {
                "path": "$.userProp1",
                "op": "=",
                "value": "abc"
            }
        ]
    }

    Sample_1()


if __name__ == '__main__':
    main()
